packages/api/lambdas/write-db-dlq-records-to-s3.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const get = require('lodash/get');
+const { s3PutObject, s3ObjectExists } = require('@cumulus/aws-client/S3');
+const { parseSQSMessageBody } = require('@cumulus/aws-client/SQS');
+
+async function handler(event, _) {
+  if (!process.env.system_bucket) throw new Error('System bucket env var is required.');
+  if (!process.env.stackName) throw new Error('Could not determine archive path as stackName env var is undefined.');
+  const sqsMessages = get(event, 'Records', []);
+  /* eslint-disable no-restricted-syntax, no-await-in-loop */
+  for (const message of sqsMessages) {
```

Comment:
Yes, because parallel processing may bring unexpected behavior if two messages from the same execution are in the batch. Although now that you mention, doesn't doing a `HEAD` introduce S3 eventual consistency issues? They would be _worse_ in a parallel `.map` process, but it's something I may need to test.

Reply:
Yeah, I think that would result in an eventual consistency issue where the second message would still get a `false` on s3ObjectExists, so I think I will refactor to group by execution ID and only check once.

Reply:
Refactored to use a `uuidv4` suffix instead.